etcdctlv3/command/compaction_command.go

```diff
@@ -43,6 +43,10 @@ func compactionCommandFunc(cmd *cobra.Command, args []string) {
 		ExitWithError(ExitError, err)
 	}
 
-	req := &pb.CompactionRequest{Revision: rev}
-	mustClient(cmd).KV.Compact(context.Background(), req)
+	c := mustClient(cmd)
+	if cerr := clientv3.NewKV(c).Compact(rev); cerr != nil {
+		fmt.Fprintf(os.Stderr, "failed to compact (%v)\n", cerr)
```

Comment:
The RPC error type is already in the code. Switched to using ExitWithError and fixed the client error handling.